README.md

```diff
@@ -1,35 +1,251 @@
-[ESLint](https://github.com/codacy/codacy-eslint) and
-[Pylint](https://github.com/codacy/codacy-pylint).
+Framework to help integration with external analysis tools at Codacy.
+
+For more details and examples of tools that use this project, you can check:
+* [PMDJava](https://github.com/codacy/codacy-pmdjava)
+* [ESLint](https://github.com/codacy/codacy-eslint)
+* [Pylint](https://github.com/codacy/codacy-pylint)
 
 ## Usage
 
 Add to your SBT dependencies:
 
 ```scala
-"com.codacy" % "codacy-engine-scala-seed_2.12" % "<VERSION>"
+"com.codacy" %% "codacy-engine-scala-seed" % "<VERSION>"
 ```
 
-You shouldn't worry about the library itself, we use it as a core in our tools,
-and everything is well explained in our Docs section.
-
 ## Docs
 
-[Docker Docs](http://docs.codacy.com/v1.5/docs/tool-developer-guide)
+### How to integrate an external analysis tool on Codacy
+
+By creating a docker and writing code to handle the tool invocation and output,
+you can integrate the tool of your choice on Codacy!
+
+> To know more about dockers, and how to write a docker file please refer to [https://docs.docker.com/reference/builder/](https://docs.docker.com/reference/builder/)
+
+In this tutorial, we explain how you can integrate an analysis tool of your choice in Codacy.
+You can check the code of an already implemented tool and, if you wish, fork it to start your own.
+You are free to modify and use it for your own tools.
+
+#### Requirements
+
+* Docker definition with the tool you want to integrate
+* Define the documentation for the patterns provided by the tool
+
+#### Assumptions and Behaviour
+
+* To run the tool we provide the configuration file, `/.codacyrc`, with the language to run and optional parameters your tool might need.
+* The source code to be analysed will be located in `/src`, meaning that when provided in the configuration, the file paths are relative to `/src`.
+
+* **.codacyrc**
+  * **files:** Files to be analysed (their path is relative to `/src`)
+  * **tools:** Array of tools
+    * **name:** Unique identifier of the tool
+    * **patterns:** Array of patterns that must be checked
+      * **patternId:** Unique identifier of the pattern
+      * **parameters:** Parameters of the pattern
+        * **name:** Unique identifier of the parameter
+        * **value:** Value to be used as parameter value
+
+```json
+{
+  "files" : ["foo/bar/baz.js", "foo2/bar/baz.php"],
+  "tools":[
+    {
+      "name":"jshint",
+      "patterns":[
+        {
+          "patternId":"latedef",
+          "parameters":[
+            {
+              "name":"latedef",
+              "value":"vars"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+Regarding to the configuration file, the tool should have different behaviours for the following situations:
+* If /.codacyrc exists and has files and patterns, use them to run.
+* If /.codacyrc exists and has only files and no patterns, run only for those files.
```

Comment:
You're missing the case "If /.codacyrc exists and only has patterns and no files..."